descqa/AngularCorrelation.py

```diff
@@ -0,0 +1,176 @@
+from __future__ import print_function, division, unicode_literals, absolute_import
+import os
+import math
+try:
+    from itertools import zip_longest
+except ImportError:
+    from itertools import izip_longest as zip_longest
+
+import numpy as np
+from GCR import GCRQuery
+
+from .base import BaseValidationTest, TestResult
+from .plotting import plt
+import treecorr
+
+__all__ = ['AngularCorrelation']
+
+
+
+
+class AngularCorrelation(BaseValidationTest):
+    """
+    Validation test to show 2pt correlation function of ProtoDC2 with SDSS
+    """
+    def __init__(self, band='r', survey='sdss', observation='', 
+                 RandomFactor=10, **kwargs):
+
+        #catalog quantities
+        self.mag_field = 'mag_{}_{}'.format(band, survey)
+        self.band = band
+        self.redshift_max = 0.3
+
+        #validation data
+        self.validation_data = {}
```

Comment:
Should we put `validation_data` in `self`? It seems to change constantly in the loop